src/tools/kata-ctl/src/main.rs

```diff
@@ -3,122 +3,35 @@
-            SubCommand::with_name("direct-volume")
-                .about("directly assign a volume to Kata Containers to manage"),
-        )
-        .subcommand(SubCommand::with_name("env").about("display settings. Default to TOML"))
-        .subcommand(SubCommand::with_name("exec").about("enter into guest by debug console"))
-        .subcommand(SubCommand::with_name("factory").about("manage vm factory"))
-        .subcommand(
-            SubCommand::with_name("help").about("shows a list of commands or help for one command"),
-        )
-        .subcommand(SubCommand::with_name("iptables").about(""))
-        .subcommand(
-            SubCommand::with_name("metrics")
-                .about("gather metrics associated with infrastructure used to run a sandbox"),
-        )
-        .subcommand(SubCommand::with_name("version").about("display version details"));
-
-    let args = app.get_matches();
-
-    let subcmd = args
-        .subcommand_name()
-        .ok_or_else(|| anyhow!("need sub-command"))?;
-
-    match subcmd {
-        "check" => {
-            match run_checks(args) {
-                Ok(_result) => println!("check may not be fully implemented"),
-                Err(err) => println!("{}", err),
-            }
-            Ok(())
-        }
-        "direct-volume" => {
-            unimplemented!("Not implemented");
-        }
-        "env" => {
-            unimplemented!("Not implemented");
-        }
-        "exec" => {
-            unimplemented!("Not implemented");
-        }
-        "factory" => {
-            unimplemented!("Not implemented");
-        }
-        "help" => {
-            unimplemented!("Not implemented");
-        }
-        "iptables" => {
-            unimplemented!("Not implemented");
-        }
-        "metrics" => {
-            unimplemented!("Not implemented");
-        }
-        "version" => {
-            println!("{} version {} (type: rust)", NAME, version);
-            Ok(())
-        }
-        _ => return Err(anyhow!(format!("invalid sub-command: {:?}", subcmd))),
-    }
-}
+use ops::check_ops::handle_check_command;
+use ops::check_ops::handle_check_volume;
+use ops::check_ops::handle_env;
+use ops::check_ops::handle_exec;
+use ops::check_ops::handle_factory;
+use ops::check_ops::handle_iptables;
+use ops::check_ops::handle_metrics;
+use ops::check_ops::handle_version;
 
 fn main() {
-    if let Err(e) = real_main() {
-        eprintln!("ERROR: {:#?}", e);
-        exit(1);
-    }
+    let args = KataCtlArgs::parse();
+
+    match args.arguments {
+        Arguments::Check(checkcmd) => handle_check_command(checkcmd),
+        Arguments::DirectVolume => handle_check_volume(),
+        Arguments::Env => handle_env(),
+        Arguments::Exec => handle_exec(),
+        Arguments::Factory => handle_factory(),
+        Arguments::Iptables(args) => handle_iptables(args),
+        Arguments::Metrics(args) => handle_metrics(args),
+        Arguments::Version => handle_version(),
+    };
```

Comment:
But we'll need to remove this semi-colon:

```suggestion
    }
```